ethcore/src/miner/miner.rs

```diff
@@ -419,7 +420,15 @@ impl Miner {
 		for tx in transactions {
 			let hash = tx.hash();
 			let start = Instant::now();
-			let result = open_block.push_transaction(tx, None);
+			// Check whether transaction type is allowed for sender
+			let result = match self.engine.machine().verify_transaction(&tx, open_block.header(), chain.as_block_chain_client()) {
```

Comment:
I guess `open_block`'s parent state always has to be available in the client, so this is fine.